src/server/journal/streamer.cc

```diff
@@ -123,12 +127,14 @@ void RestoreStreamer::WriteBucket(PrimeTable::bucket_iterator it) {
 
   {
     FiberAtomicGuard fg;  // Can't switch fibers because that could invalidate iterator
-
+    string key_buffer;    // we can reuse it
     while (!it.is_done()) {
       const auto& pv = it->second;
 
-      string key_buffer;
       string_view key = it->first.GetSlice(&key_buffer);
+      if (!ShouldWrite(key)) {
```

Comment:
If no slot from the bucket was written as the bucket did not contain any slot data we should not call NotifyWritten below